types/src/blob.rs

```diff
@@ -722,6 +721,62 @@ mod tests {
         .unwrap()
     }
 
+    #[test]
+    fn create_new_blob_unsigned() {
+        let my_namespace = Namespace::new_v0(&[1, 2, 3, 4, 5]).expect("Invalid namespace");
+        let blob_unsigned = Blob::new(
+            my_namespace,
+            b"some data to store on blockchain".to_vec(),
+            None,
+            AppVersion::V2,
+        )
+        .expect("Failed to create a blob");
+
+        let actual_unsigned = serde_json::to_string_pretty(&blob_unsigned).unwrap();
+        let expected_unsigned = indoc::indoc! {r#"{
+          "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQIDBAU=",
+          "data": "c29tZSBkYXRhIHRvIHN0b3JlIG9uIGJsb2NrY2hhaW4=",
+          "share_version": 0,
+          "commitment": "m0A4feU6Fqd5Zy9td3M7lntG8A3PKqe6YdugmAsWz28=",
+          "index": -1,
+          "signer": null
+        }"#};
+
+        assert_eq!(
+            serde_json::from_str::<serde_json::Value>(&actual_unsigned).unwrap(),
+            serde_json::from_str::<serde_json::Value>(expected_unsigned).unwrap()
+        );
+    }
+
+    #[test]
+    fn create_new_blob_signed() {
+        let my_namespace = Namespace::new_v0(&[1, 2, 3, 4, 5]).expect("Invalid namespace");
+        let signer: AccAddress = "celestia1377k5an3f94v6wyaceu0cf4nq6gk2jtpc46g7h"
+            .parse()
+            .unwrap();
+        let blob_signed = Blob::new(
+            my_namespace,
+            b"some data to store on blockchain".to_vec(),
+            Some(signer),
+            AppVersion::V5,
+        )
+        .expect("Failed to create a signed blob");
+
+        let actual_signed = serde_json::to_string_pretty(&blob_signed).unwrap();
+        let expected_signed = indoc::indoc! {r#"{
+          "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQIDBAU=",
+          "data": "c29tZSBkYXRhIHRvIHN0b3JlIG9uIGJsb2NrY2hhaW4=",
+          "share_version": 1,
+          "commitment": "AUpLKHYnlrfK0cX6DcGyryFMld1nJia+cjkCwXFTFgA=",
+          "index": -1,
+          "signer": "j71qdnFJas04ncZ4/CazBpFlSWE="
+        }"#};
+
+        assert_eq!(
+            serde_json::from_str::<serde_json::Value>(&actual_signed).unwrap(),
+            serde_json::from_str::<serde_json::Value>(expected_signed).unwrap()
+        );
+    }
```

Comment:
Main thrust of changes here, reflected in all doc comments that are tested with `cargo test --doc` passing.